src/components/pricing/pricing-widget/index.tsx

```diff
@@ -71,6 +73,26 @@ const PricingWidget: FunctionComponent<{}> = () => {
     })
 
     if (emailIsValid(viewer?.email)) {
+      const {hasProAccess} = await axios
```

Comment:
one. i e to have would be to move calls like this into lib/users versus raw axios calls. same thing ultimately just jive since we probably make this call in multiple places

Reply:
here’s another spot where we are making this check. on the pricing page we also ask for an email immediately and check what they enter there. 

https://github.com/eggheadio/egghead-next/blob/f84a9c2219f8e11d01a80da759d54385b79345c1/src/pages/pricing/email/index.tsx#L44

are the “widgets” not doing that prior to this pr?

(there are three spots in the app that make this call)

Reply:
> move calls like this into lib/users versus raw axios calls.

for sure, I want to make another pass that consolidates these pro access checks and moves the axios call into `/lib`. we've "favored duplication over the wrong abstraction" enough and I think we can come up with something shared now.